### ISSUE TITLE
Confusion matrix for CRNN

### ISSUE THREAD

**Issue by ahmedmazari-dhatim** [meijieru/crnn.pytorch#42]
Hello,

l'm thinking about making confusion matrix for CRNN. However the output classes are variable sequences and not as standard output classes as  0 1 2 .. or  A B C . 
Any idea of how making confusion matrix for CRNN.

Thank you

**Comment by meijieru** [comment]
Confusion matrix play a more important role in classification.
For outputs like text, other metrics, editing distance for example, are more meaningful.

**Comment by ahmedmazari-dhatim** [comment]
@meijieru thank you for your answer.
We can say that sequence prediction is a classification problem, isn't it ?
Because we have an input which is an image that contains a sequence of character and its label is that sequence. Can't we use confusion matrix in this case ?
otherwise, what do you propose ?

**Comment by rremani** [comment]
@ahmedmazari-dhatim 
I went through a paper of alex graves, he gave this formulae
100 * (1- (insertion+deletions+subsitutions)/total number of words))
Here, words with insertion, deletion & substitution are counted and accuracy is obtained.

**Comment by ahmedmazari-dhatim** [comment]
Hi @rremani ,

Do you mind sharing with us this article ?

l'm not sure to understand  the relationship between confusion matrix and the the formula 100 * (1- (insertion+deletions+subsitutions)/total number of words)) ?

Thank you

**Comment by rremani** [comment]
sure, It's a chapter from a book, 'Neural Networks for Handwriting Recognition'.
DOI number: doi.org/10.1007/978-3-642-24049-2_2
You can get this through sci-hub ;)
cheers

**Comment by ahmedmazari-dhatim** [comment]
http://doi.org.sci-hub.io/10.1007/978-3-642-24049-2_2 is not working

**Comment by rremani** [comment]
[liwicki2012.pdf](https://github.com/meijieru/crnn.pytorch/files/1229163/liwicki2012.pdf)

**Comment by ahmedmazari-dhatim** [comment]
Thank you
